### PR TITLE
Add a robot for the `ChessBoard` composable

### PR DESCRIPTION
**Pull request by alexandrepiveteau**
This is part of the refactoring required of #153. This pull request adds a `ChessBoardRobot`, and updates some existing chessboard tests to use it.

### CODE REVIEW

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt [hunk 1]
```diff
@@ -27,3 +29,14 @@ fun SemanticsNode.getBoundsInRoot(): DpRect =
...
+
+/**
+ * Returns true iff the [Offset] is contained within this [Rect] bounds.
+ *
+ * @receiver the bounds of the [Rect].
+ * @param offset the offset that we're checking some bounds for.
+ * @return true if the offset is contained.
+ */
+operator fun Rect.contains(offset: Offset): Boolean {
+  return offset.x in left..right && offset.y in top..bottom
```
Comment (barmettlerl) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt:41]: What is the `..`? Never saw this before

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ChessBoardRobot.kt [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+   * Moves the pointer with the given id to this cell.
+   *
+   * @param x the x coordinate of the target cell.
+   * @param y the y coordinate of the target cell.
+   * @param pointerId the identifier of the pointer.
+   */
+  fun moveTo(x: Int, y: Int, pointerId: Int = 0)
+
+  /**
+   * Moves the pointer with the given id by a relative amount of cells.
```
Comment (barmettlerl) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ChessBoardRobot.kt:41]: Question: what do you mean by pointer?

**Review by barmettlerl — APPROVED** [review]

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt (hunk 1 above)
Comment (alexandrepiveteau) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt:41]: This is the `rangeTo` operator ([reference](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/range-to.html)).

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ChessBoardRobot.kt (hunk 2 above)
Comment (alexandrepiveteau) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ChessBoardRobot.kt:41]: On multi-touch devices, gestures are composed of multiple pointers (usually the fingers of the user). This terminology is used by Android gesture system (see the documentation for [`MotionEvent`](https://developer.android.com/reference/android/view/MotionEvent) for instance). Compose uses the same terminology.

The default pointer has value 0. If you look into Jetpack Compose's `SemanticsNodeInteraction.performTouchInput { }` method, you'll see that gestures are performed with this pointer by default, but allow the user to specify another pointer identifier. The `ChessBoardRobot` essentially allows the same behaviour, so we can test multi-touch gestures too.

**Review by Fouad-sys — APPROVED** [review]
LGTM!

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt (hunk 1 above)
Comment (Fouad-sys) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/Layout.kt:41]: Was going to ask the same question, thanks for clearing that up!

**Pull request merged**